### DISCUSSION TITLE
How to consume the local build on Linux

### DISCUSSION THREAD

**Discussion by jane-mao** [AvaloniaUI/Avalonia#14030] [Q&A] [answered]
I have built Avalonia 11.0.6 on Windows and Linux according the below guides:
https://github.com/AvaloniaUI/Avalonia/blob/master/Documentation/build.md
https://github.com/AvaloniaUI/Avalonia/blob/master/azure-pipelines.yml

There are a bunch of .nuget packages under the sub-folder artifacts/nuget, and I installed all of these packages with nuget command, then run our Avalonia sample code on Linux, but it prompted some errors as below:
```
error CS0012: The type 'IPlatformHandle' is defined in an assembly that is not referenced. You must add a reference to assembly 'Avalonia.Base, Version=11.0.0.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b'.
error CS0012: The type 'IApplicationLifetime' is defined in an assembly that is not referenced. You must add a reference to assembly 'Avalonia.Controls, Version=11.0.0.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b'.
```
Looks the assembly Avalonia.Controls and Avalonia.Base were not found.
Is there any guide to consume the local build?

Thanks.

**Comment by thevortexcloud** [comment]
The version of those assemblies is suspicious if you are trying to run against `11.0.6`. Do you have some other existing package reference?

**Reply by jane-mao** [reply]
Sorry, I did not get you point:
> Do you have some other existing package reference?

I used the following command to build Avalonia on Linux:`./build.sh --target Package --configuration Release`.
The generated .nuget packages are listed in the attached screenshot.

Thanks.
![Uploading 2023-12-25 14.51.08.png…]()

**Reply by thevortexcloud** [reply, marked as answer]
It's complaining that it can't find `11.0.0.0` which as far as I am aware, is not the latest version of those assemblies (I could be wrong though). If you have another package in your consumer project that explicitly needs those assemblies then it wont build without them.

Your screenshot however did not upload correctly.

**Reply by jane-mao** [reply]
It has been fixed, thanks.
Another question, the Avalonia.*.nupkg can be built on Mac and used on Windows/Linux, correct?

**Reply by thevortexcloud** [reply]
It is not possible to build _all_ of Avalonia outside of Windows. But you should be able to build most of it. For those that do work it _should_ work fine. I have never tried it though. .NET in general does not really care what platform builds the code, it all gets compiled to the same code and then the platform specific runtime converts it to native code for the platform. All the .NET packaging tools work Mac as far as I am aware too.

**Comment by jane-mao** [comment]
Re-upload the screenshot <img width="1526" alt="2023-12-25 14 51 08" src="https://github.com/AvaloniaUI/Avalonia/assets/62095850/ae3bc971-db4a-4ea7-b31f-5df3481c5cab">

**Comment by timunie** [comment]
You may need to install the packages in the other project as well if you have more than one prj.

**Reply by jane-mao** [reply]
Yes, thanks.
